src/sparseml/evaluation/integrations/lm_evaluation_harness.py

```diff
@@ -13,36 +13,46 @@
 # limitations under the License.
 
 import logging
-from typing import Any, Dict, List, Optional
-
-from pydantic import BaseModel, Field
+from typing import Any, Dict, List
 
 from sparseml.evaluation.registry import SparseMLEvaluationRegistry
-from sparseml.transformers.utils.sparse_model import SparseAutoModelForCausalLM
 from sparsezoo.evaluation.results import Dataset, Evaluation, Metric, Result
 
 
 try:
     from lm_eval import evaluator, tasks, utils
     from lm_eval.models.huggingface import HFLM
-except ImportError as err:
+except ImportError as import_error:
     HFLM = object
-    raise Exception(
-        "package `lm_eval` is not installed. "
-        "Please install it via "
-        "`pip install lm-eval==0.4.0"
-    ) from err
-
-__all__ = ["lm_eval_harness", "SparseMLLM", "LMEvalHarnessEvaluatorInputSchema"]
+    raise ImportError(
+        "package `lm_eval` not found. Please install it via "
+        "`pip install git+https://github.com/EleutherAI/lm-evaluation-harness.git@e0eda4d`"  # noqa: E501
+    ) from import_error
+try:
+    # This needs to be imported after lm_eval to ensure right transformers
+    # version is installed for SparseML
+    from sparseml.transformers.utils.sparse_config import SparseAutoConfig
+    from sparseml.transformers.utils.sparse_model import SparseAutoModelForCausalLM
+except ImportError as import_error:
+    raise ImportError(
+        "Install sparseml supported dependencies for lm-eval integration by running "
+        "`pip uninstall transformers && pip install sparseml[transformers,torch]`"
```

Comment:
soon to be deprecated because of forkless transformers support

Reply:
Still needed for now, will keep in mind  though 💪🏽